### PR TITLE
WIP: Database Rewrite

### PR DESCRIPTION
**Pull request by mboecker**
This adds the script to re-write the database to the format we developed.
It also breaks the server.

TODO: 

- [x] Add eval-measures to reuslting database
- [x] Add all task ids

### CODE REVIEW

**Comment by jakob-r** [comment]
why don't you use `dbWriteTable` to put the new table into the DB?

**Comment by mboecker** [comment]
That function didn't work for me. "table already exists" and "table doesnt exist"-errors, depending of whether I create the table first or not.

File: docker/preparation/reformat_db.R [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+  return(t)
+}
+
+#' Return a list of parameter definitions. This list contains every necessary parameter for the given algorithm.
+#'
+#' @param task_id This is `task_id` from the table `run` in the database.
+#'
+#' @return A named list with one entry for each needed parameter.
+get_params_for_algo = function(algo_name) {
```
Comment (jakob-r) [inline docker/preparation/reformat_db.R]: This function basically just takes `parameter_ranges[[algo_name]]` and converts it to a list that can be serialized to JSON. There should be no reason to call it if the serialization will not be applied later on. You can just write `parameter_ranges[[algo_name]]` instead.

**Comment by jakob-r** [comment]
Well this function should work. It seems like multiple people had problems with this function. Does the table name contain capital letters? 

Maybe we should try to fix this bug instead of writing our own function. What is the output of `dbExistsTable()` and `dbListTables()`? The input has to be an exact match of the latter.

**Comment by mboecker** [comment]
I think I changed all the points you criticised. All thats missing now is the actual API I think.

File: docker/omlbotlookup/data_access.R [hunk 2]
```diff
@@ -205,7 +205,10 @@ get_nearest_setup = function(algo_id, task_id, parameters) {
   mins = sapply(table[, ..np, drop = FALSE], min)
   maxs = sapply(table[, ..np, drop = FALSE], max)
   table.scaled = scale(table[, ..np, drop = FALSE], center = mins, scale = maxs - mins)
-  table[, np] = table.scaled
```
Comment (mboecker) [inline docker/omlbotlookup/data_access.R]:
```
Warnung in `[<-.data.table`(`*tmp*`, , np, value = c(0.0650325162581291, 
  4 column matrix RHS of := will be treated as one vector
Warnung in `[<-.data.table`(`*tmp*`, , np, value = c(0.0650325162581291, 
  Supplied 46544 items to be assigned to 11636 items of column 'num.trees' (34908 unused)
Warnung in `[<-.data.table`(`*tmp*`, , np, value = c(0.0650325162581291, 
  4 column matrix RHS of := will be treated as one vector
Warnung in `[<-.data.table`(`*tmp*`, , np, value = c(0.0650325162581291, 
  Supplied 46544 items to be assigned to 11636 items of column 'sample.fraction' (34908 unused)
Warnung in `[<-.data.table`(`*tmp*`, , np, value = c(0.0650325162581291, 
  4 column matrix RHS of := will be treated as one vector
Warnung in `[<-.data.table`(`*tmp*`, , np, value = c(0.0650325162581291, 
  Supplied 46544 items to be assigned to 11636 items of column 'mtry' (34908 unused)
Warnung in `[<-.data.table`(`*tmp*`, , np, value = c(0.0650325162581291, 
  4 column matrix RHS of := will be treated as one vector
Warnung in `[<-.data.table`(`*tmp*`, , np, value = c(0.0650325162581291, 
  Supplied 46544 items to be assigned to 11636 items of column 'min.node.size' (34908 unused)
Warnung in `[<-.data.table`(`*tmp*`, , np, value = c(0.0650325162581291, 
  Coerced 'double' RHS to 'integer' to match the column's type; may have truncated precision. Either change the target column ['num.trees'] to 'double' first (by creating a new 'double' vector length 11636 (nrows of entire table) and assign that; i.e. 'replace' column), or coerce RHS to 'integer' (e.g. 1L, NA_[real|integer]_, as.*, etc) to make your intent clear and for speed. Or, set the column type correctly up front when you create the table and stick to it, please.
Warnung in `[<-.data.table`(`*tmp*`, , np, value = c(0.0650325162581291, 
  Coerced 'double' RHS to 'integer' to match the column's type; may have truncated precision. Either change the target column ['mtry'] to 'double' first (by creating a new 'double' vector length 11636 (nrows of entire table) and assign that; i.e. 'replace' column), or coerce RHS to 'integer' (e.g. 1L, NA_[real|integer]_, as.*, etc) to make your intent clear and for speed. Or, set the column type correctly up front when you create the table and stick to it, please.

```

Comment (mboecker) [reply docker/omlbotlookup/data_access.R]: Obwohl die Zeilen und spaltenanzahl auf beiden seiten gleich war und die spalten auch gleich hießen. seltsam.

**Pull request merged**